### PR TITLE
[react-select] Add generic argument for custom select components props

### PR DESCRIPTION
**Pull request by szymkab**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/34959

### CODE REVIEW

File: types/react-select/lib/components/MultiValue.d.ts [hunk 1]
```diff
@@ -19,36 +20,36 @@ export type MultiValueProps<OptionType> = CommonProps<OptionType> &{
...
-export type MultiValueRemoveProps<OptionType> = CommonProps<OptionType> & {
+export type MultiValueRemoveProps<OptionType, SelectComponentsProps = DefaultSelectComponentsProps> = CommonProps<OptionType, SelectComponentsProps> & {
   children: ReactNode,
   innerProps: {
     className: string,
     onTouchEnd: (event: any) => void,
     onClick: (event: any) => void,
     onMouseDown: (event: any) => void,
   },
-  selectProps: any,
```
Comment (claasahl) [inline types/react-select/lib/components/MultiValue.d.ts]: Was this intentionally removed? Or should this have been something like `selectProps: SelectProps<OptionType, SelectComponentsProps>;`?

**Review by claasahl — CHANGES_REQUESTED** [review]
Thanks for your contribution, @szymkab 

Noticed that `selectProps` in `MultiValueRemoveProps` was removed. Was this intentional?

https://github.com/szymkab/DefinitelyTyped/blob/c6a47697c6327e5ed2881d99d266aa0cda1d23a0/types/react-select/lib/components/MultiValue.d.ts#L36-L44

File: types/react-select/lib/components/MultiValue.d.ts (hunk 1 above)
Comment (szymkab) [reply types/react-select/lib/components/MultiValue.d.ts]: Yes, I've removed it because `selectProps` field is in `CommonProps`.

**Review by claasahl — COMMENTED** [review]
LGTM

**Comment by claasahl** [comment]
Sorry for the delay @szymkab

If you open a new PR, I'll approve it asap

**Pull request closed without merging**